src/display/canvas.js

```diff
@@ -478,6 +472,7 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
     this.smaskCounter = 0;
     this.tempSMask = null;
     this.cachedCanvases = new CachedCanvases();
+    this.DOMcanvasFactory = new DOMcanvasFactory();
```

Comment:
Furthermore, I think that you also need to initialize the `DOMCanvasFactory` *before* `CachedCanvases`, since you'll probably need to do `this.cachedCanvases = new CachedCanvases(this.canvasFactory);`.
Please refer to https://github.com/mozilla/pdf.js/pull/8002/files#r98327854.